### ISSUE TITLE
Error when creating a github_branch_protection_v3

### ISSUE THREAD

**Issue by gilacost** [integrations/terraform-provider-github#1451]
## Terraform Version
```
Terraform v1.3.4
on darwin_amd64
+ provider registry.terraform.io/integrations/github v5.12.0
```

## Affected Resource
* github_branch_protection_v3

## Terraform Configuration Files

```hcl
terraform {
  required_providers {
    github = {
      source  = "integrations/github"
      version = "5.12.0"
    }
  }

  required_version = ">= 1.0"
}

provider "github" {
  owner = local.owner
}

locals {
  owner = "PepOps"
  repos = [
    {
      name         = "test",
      description  = "a  description"
      context_list = ["Lint Code Base"]
    }
  ]
}

resource "github_repository" "main" {
  allow_auto_merge            = true
  allow_merge_commit          = false
  allow_rebase_merge          = false
  allow_squash_merge          = true
  squash_merge_commit_title   = "PR_TITLE"
  squash_merge_commit_message = "COMMIT_MESSAGES"
  delete_branch_on_merge      = true
  description                 = local.repos[0].description
  has_issues                  = true
  has_projects                = true
  name                        = local.repos[0].name
  topics                      = []
  visibility                  = "private"
  vulnerability_alerts        = true
  allow_update_branch         = true
}

resource "github_branch_protection_v3" "main" {
  branch                          = "main"
  enforce_admins                  = false
  repository                      = "test"
  require_conversation_resolution = false
  require_signed_commits          = false

  required_pull_request_reviews {
    dismiss_stale_reviews           = false
    dismissal_teams                 = []
    dismissal_users                 = []
    include_admins                  = false
    require_code_owner_reviews      = false
    required_approving_review_count = 1
  }

  required_status_checks {
    contexts = [
      "Lint Code Base",
    ]
    include_admins = false
    strict         = true
  }
}

```

## Debug Output

https://gist.githubusercontent.com/gilacost/91ed474e824d1efe21881bde22e8cd97/raw/0750a86e2ddb9b45ef9962fa74f2d2ad3fbf247d/tf.log

## Panic Output

None

## Expected Behaviour

`github_branch_protection_v3` is created successfully.

## Actual Behaviour
```bash
╷
│ Error: PUT https://api.github.com/repos/BeamOps/book/branches/main/protection: 422 Invalid request.
│ 
│ No subschema in "anyOf" matched.
│ For 'properties/checks', nil is not an array.
│ Not all subschemas of "allOf" matched.
│ For 'anyOf/1', {"strict"=>true, "contexts"=>["Lint Code Base"], "checks"=>nil} is not a null. []
│ 
│   with github_branch_protection_v3.main,
│   on main.tf line 48, in resource "github_branch_protection_v3" "main":
│   48: resource "github_branch_protection_v3" "main" {
│ 
╵
```
## Steps to Reproduce

    1. `terraform apply`

## Important Factoids

None

## References

None

**Comment by TheQueenIsDead** [comment]
I'm not sure this issue needs to be open as it duplicates https://github.com/integrations/terraform-provider-github/issues/1307 and somewhat https://github.com/integrations/terraform-provider-github/issues/1147, though believe me I know the frustration!

**Comment by kfcampbell** [comment]
@TheQueenIsDead is right; I'm going to close this. Please subscribe to the other issue(s) for updates, and I'll make time to review the proposed fix in #1415 soon.
